File: 2-2.lua
io.input('2.txt')
local depth = 0
local horiz = 0
local aim = 0
local direction, magnitude

for line in io.lines() do
    direction, magnitude = string.match(line, '(%a+) (%d+)')

    if direction == 'up' then aim = aim - magnitude end
    if direction == 'down' then aim = aim + magnitude end
    if direction == 'forward' then
        horiz = horiz + magnitude
        depth = depth + (magnitude * aim)
    end
end


print(string.format("depth = %d, horiz = %d, aim = %d", depth, horiz, aim))
print(string.format("depth * horiz = %d", depth * horiz))

File: 2-2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
	FILE *file = fopen("2.txt", "r");
	char command[8];
	int magnitude;

	int depth = 0;
	int horiz = 0;
	int aim = 0;
	while (fscanf(file, "%s %d\n", command, &magnitude) != EOF) {
		if (strcmp(command, "up") == 0)
			aim -= magnitude;
		if (strcmp(command, "down") == 0)
			aim += magnitude;
		if (strcmp(command, "forward") == 0) {
			horiz += magnitude;
			depth += magnitude * aim;
		}
	}
	printf("depth = %d, horiz = %d, aim = %d\n", depth, horiz, aim);
	printf("=> %d\n", depth * horiz);
	fclose(file);
}
